plugins/query-insights/src/main/java/org/opensearch/plugin/insights/settings/QueryInsightsSettings.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.insights.settings;
+
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.plugin.insights.core.exporter.QueryInsightsExporterType;
+
+import java.util.Locale;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Settings for Query Insights Plugin
+ *
+ * @opensearch.api
+ * @opensearch.experimental
+ */
+public class QueryInsightsSettings {
+    /**
+     * Default Values and Settings
+     */
+    public static final TimeValue MAX_WINDOW_SIZE = new TimeValue(21600, TimeUnit.SECONDS);
+    /** Default N size for top N queries */
+    public static final int MAX_N_SIZE = 100;
+    /** Default min export interval for Query Insights exporters */
+    public static final TimeValue MIN_EXPORT_INTERVAL = new TimeValue(1, TimeUnit.SECONDS);
+    /** Default local index mapping for top n queries records */
+    public static final String DEFAULT_LOCAL_INDEX_MAPPING = "mappings/top_n_queries_record.json";
```

Comment:
does it needs to be public?

Reply:
Yes, I want to gather all settings for Query Insights into this setting class, they need to be public in order for other classes to access.